python/ray/tune/impl/tuner_internal.py

```diff
@@ -145,6 +137,16 @@ def __init__(
         if not trainable:
             raise TuneError("You need to provide a trainable to tune.")
 
+        if self._entrypoint == AirEntrypoint.TUNER and not isinstance(
+            self._run_config, ray.tune.RunConfig
+        ):
```

Comment:
I did this instead of "casting" `ray.train.RunConfig` -> `ray.tune.RunConfig` because doing that would still emit the warning on the restoration flow and it was complicated to fix the problem for that case too. Adding this entrypoint condition + moving it to the "start from scratch" codepath solves the problem.